### PR TITLE
plan: process in subquery correctly.

### LINKED ISSUE #2007: Seems that there is parser problem when using IN with a double "(())"
Please answer these questions before submitting your issue. Thanks!

1. What version of Go are you using (`go version`)?
1.7.3

2. What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN="/usr/local/go/bin"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/gopkg"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build945606389=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"

3. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.

```
DROP TABLE IF EXISTS t1;
DROP TABLE IF EXISTS t2;
DROP TABLE IF EXISTS t1xt2;

CREATE TABLE t1 (
  id_1 int(5) NOT NULL,
  t varchar(4) DEFAULT NULL
);

CREATE TABLE t2 (
  id_2 int(5) NOT NULL,
  t varchar(4) DEFAULT NULL
);

CREATE TABLE t1xt2 (
  id_1 int(5) NOT NULL,
  id_2 int(5) NOT NULL
);

INSERT INTO t1 VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd');
INSERT INTO t2 VALUES (2, 'bb'), (3, 'cc'), (4, 'dd'), (12, 'aa');
INSERT INTO t1xt2 VALUES (2, 2), (3, 3), (4, 4);

SELECT DISTINCT t1.id_1 FROM t1 WHERE
(12 NOT IN (SELECT t1xt2.id_2 FROM t1xt2 WHERE t1.id_1 = t1xt2.id_1));

SELECT DISTINCT t1.id_1 FROM t1 WHERE
(12 NOT IN ((SELECT t1xt2.id_2 FROM t1xt2 where t1.id_1 = t1xt2.id_1)));

```

4. What did you expect to see?

the result of two queries should be both
```
+------+
| id_1 |
+------+
| 1    |
| 2    |
| 3    |
| 4    |
+------+

```

5. What did you see instead?
the first is the expected result;
the second is:
```
+------+
| id_1 |
+------+
| 2    |
| 3    |
| 4    |
+------+
```

### PR DESCRIPTION
**Pull request by hanfei1991**
fix #2007
For case "10 in ((select * from t))", the planner will treat in as function and select * from t as scalar function.
After fixing it, it will be treated as in sub query.
@shenli @coocood @winoros @XuHuaiyu @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: plan/expression_rewriter.go [hunk 1]
```diff
@@ -136,6 +136,24 @@ func (er *expressionRewriter) Enter(inNode ast.Node) (ast.Node, bool) {
...
+	loop:
+		for {
+			switch y := x.(type) {
+			case *ast.SubqueryExpr:
+				v.Sel = y
+				return er.handleInSubquery(v)
+			case *ast.ParenthesesExpr:
+				x = y.Expr
+			default:
+				break loop
```
Comment (zimulala) [inline plan/expression_rewriter.go]: Remove `loop` and return `inNode， false` directly.

File: plan/logical_plan_test.go [hunk 2]
```diff
@@ -589,6 +514,47 @@ func (s *testPlanSuite) TestPredicatePushDown(c *C) {
...
 }
 
+func (s *testPlanSuite) TestLogicalPlanBuilder(c *C) {
+	defer testleak.AfterTest(c)()
+	cases := []struct {
+		sql  string
+		best string
+	}{
+		{
+			// treated as in sub query.
```
Comment (zimulala) [inline plan/logical_plan_test.go]: Complete the sentence. the same as line 529 and line 534.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**